osv/models.py

```diff
@@ -429,11 +432,25 @@ def _pre_put_hook(self):  # pylint: disable=arguments-differ
         # No need to normalize if the ecosystem is supported.
         self.affected_fuzzy.extend(affected_package.versions)
       else:
-        self.affected_fuzzy.extend(
-            bug.normalize_tags(
-                _maybe_strip_repo_prefixes(
-                    affected_package.versions,
-                    [range.repo_url for range in affected_package.ranges])))
+        if (not affected_package.package.ecosystem and
+            len(affected_package.versions) > _MAX_GIT_VERSIONS_TO_INDEX):
+          # Assume that if there is no ecosystem specified, then these versions
+          # were enumerated from Git.
+          #
+          # Mitigate cases where the Git repo tag matching results in too many
+          # versions to index for Datastore.
+          # It's OK to do this because the primary intended matching mechanism
+          # for Git is via commit hash matching instead.
+          logging.info(
```

Comment:
Could we also add/append to https://ossf.github.io/osv-schema/#affecteddatabase_specific-field here so it's explicit to a record consumer that we've made this decision here?

Reply:
This has nothing to do with database_specific.

This is just an indexing side change for OSV.dev. The only visible changes for OSV producers is that more of their records will import, but they will fail to match via a version query to OSV.dev API.

Reply:
If I've read this correctly, for such records `affected.versions[]` will be completely empty (versus truncated to 5000)? I guess as an alternative to making this explicitly clear in the record itself, calling it out at https://google.github.io/osv.dev/faq/#what-does-osvdev-do-to-the-records-it-imports ?

The problem I frequently run into when reading the code that does the work is there's a lot of other "last-minute" things that go on in the pre-put hook in `models.py` that aren't immediately obvious and violate the principle of least surprise.

Reply:
No, we are simply not indexing them in `affected_fuzzy` (which is not visible to users). We are not truncating them in the OSV record.

Reply:
see https://github.com/google/osv.dev/blob/c5c3e96d0e9a0b56922bc7a32c27f0c4975396aa/osv/models.py#L274

Reply:
And also the test output file: https://github.com/google/osv.dev/blob/17ee9228492825626dc0efb1d6d4eef1075ed00c/docker/worker/testdata/UpdateTest_dont_index_too_many_git_versions.txt#L15

Reply:
And also the test output file: https://github.com/google/osv.dev/blob/17ee9228492825626dc0efb1d6d4eef1075ed00c/docker/worker/testdata/UpdateTest_dont_index_too_many_git_versions.txt#L15

Reply:
My bad, I misunderstood indexing as enumeration...